src/components/scenes/AssetSettingsScene.tsx

```diff
@@ -1,20 +1,59 @@
 import * as React from 'react'
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
 
 import { CURRENCY_SETTINGS_KEYS } from '../../constants/WalletAndCurrencyConstants'
+import { useHandler } from '../../hooks/useHandler'
+import { lstrings } from '../../locales/strings'
 import { useSelector } from '../../types/reactRedux'
 import { EdgeSceneProps } from '../../types/routerTypes'
 import { SceneWrapper } from '../common/SceneWrapper'
+import { showToast } from '../services/AirshipInstance'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
 import { SettingsTappableRow } from '../settings/SettingsTappableRow'
 import { CryptoIconUi4 } from '../ui4/CryptoIconUi4'
 
 interface Props extends EdgeSceneProps<'assetSettings'> {}
 
 export function AssetSettingsScene(props: Props) {
   const { navigation } = props
+  const theme = useTheme()
+  const styles = getStyles(theme)
+
   const account = useSelector(state => state.core.account)
+  const { currencyWallets } = account
+
+  /**
+   * Iterates through all the user's currency wallets, updates the enabled token
+   * IDs to include any newly detected tokens, and then displays a toast message
+   * indicating whether any detected tokens were enabled.
+   */
+  const handleRescanTokens = useHandler(async () => {
+    const modifiedWallets = []
```

Comment:
How is TypeScript OK with this? I would have expected `const modifiedWallets: EdgeCurrencyWallet[] = []` to be needed.

Reply:
I named it misleadingly, it's a string[]